### PR TITLE
[IMP] debt history updating on incoming partner balance updates

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

File: pos_debt_notebook/static/src/js/pos.js [hunk 1]
```diff
@@ -85,6 +85,20 @@ odoo.define('pos_debt_notebook.pos', function (require) {
...
                 limit = 0;
             }
+            if (partner_ids.length === 1 && !limit){
+                // incoming data about balance updates not contains limit
+                // this block changes limit only in case of opened debt history for a partner who have got an update
+                var client_list_screen = this.gui.screen_instances.clientlist;
+                if (client_list_screen && client_list_screen.debt_history_is_opened()){
+                    var partner = client_list_screen.new_client || this.get_client();
+                    if (partner_ids[0] === partner.id){
+                        limit = Math.min(Math.max($('#debt_history tr').length - 2, 0), partner.records_count);
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js]: Don't get logic information from UI

Comment (yelizariev) [reply pos_debt_notebook/static/src/js/pos.js]: Save history length somewhere in js model instead

File: pos_debt_notebook/static/src/js/pos.js [hunk 2]
```diff
@@ -787,7 +801,8 @@ odoo.define('pos_debt_notebook.pos', function (require) {
                 }
                 debt = this.format_currency(debt);
 
-                if (_.contains(partner_ids, partner.id)){
+                if (partner_ids.length === 1 && partner_ids[0] === partner.id){
+                    // updating only opened partner profile and deebt history
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js]: deebt

File: pos_debt_notebook/static/src/js/pos.js [hunk 3]
```diff
@@ -85,6 +85,20 @@ odoo.define('pos_debt_notebook.pos', function (require) {
...
             }
+            if (partner_ids.length === 1 && !limit){
+                // incoming data about balance updates not contains limit
+                // this block changes limit only in case of opened debt history for a partner who have got an update
+                var client_list_screen = this.gui.screen_instances.clientlist;
+                if (client_list_screen && client_list_screen.debt_history_is_opened()){
+                    var partner = client_list_screen.new_client || this.get_client();
+                    if (partner_ids[0] === partner.id){
+                        limit = Math.min(Math.max($('#debt_history tr').length - 2, 0), partner.records_count);
+                        if (!limit && !partner.records_count){
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js]: Should it be just ``if (!limit)`` ?

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Review by Railia — APPROVED** [review]

**Review by yelizariev — APPROVED** [review]
code looks good

**Pull request merged**